Sources/SPMBuildCore/BinaryTarget+Extensions.swift

```diff
@@ -97,6 +97,14 @@ fileprivate extension Triple.OS {
             return nil // XCFrameworks do not support any of these platforms today.
```

Comment:
`asXCFrameworkPlatformString` needs to move to an extension on `Triple` rather than `Triple.OS`, as you also need access to the triple environment to compute the full thing. For example:

```swift
fileprivate extension Triple {
    var asXCFrameworkPlatformString: (String, String)? {
        switch (os, abi) {
        case (.macOS, .unknown):
            return ("macos", "")
        case (.ios, .other(name: "macabi")), (.ios, .other(name: "maccatalyst")):
            return ("ios", "maccatalyst")
        case (.ios, .unknown):
            return ("ios", "")
        case (.ios, .other(name: "simulator")):
            return ("ios", "simulator")
        case (.tvos, .unknown):
            return ("tvos", "")
        case (.tvos, .other(name: "simulator")):
            return ("tvos", "simulator")
        case (.watchos, .unknown):
            return ("watchos", "")
        case (.watchos, .other(name: "simulator")):
            return ("watchos", "simulator")
        case (.driverkit, .unknown):
            return ("driverkit", "")
        default:
            return nil
        }
    }
}
```

Then XCFrameworkMetadata.Library needs to gain a new `platformVariant` property (whose CodingKey is `"SupportedPlatformVariant"` and might not be present), and finally, the the code in `parseXCFrameworks` needs to be adjusted to use that new property:

```swift
        guard let library = metadata.libraries.first(where: {
            ($0.platform, $0.environment) == triple.os.asXCFrameworkPlatformString &&
            $0.architectures.contains(triple.arch.rawValue)
        }) else {
            return []
        }
```